www/nikd01/cv04/utils/utils.php

```diff
@@ -0,0 +1,86 @@
+<?php
+function fetchUsers($filename): array
+{
+    $users = [];
+    $file = fopen($filename, 'r');
+
+    if ($file) {
+        while (($line = fgetcsv($file, 0, ";")) !== FALSE) {
+            if (count($line) > 1) { // Make sure the line has at least 2 elements
+                $users[$line[1]] = [
+                    'name' => $line[0],
+                    'email' => $line[1],
+                    'password' => $line[2]
+                ];
+            }
+        }
+        fclose($file);
+    }
+
+    return $users;
+}
+
+function fetchUser($filename, $email): ?array
+{
+    $file = fopen($filename, 'r');
+
+    if ($file) {
+        while (($line = fgetcsv($file, 0, ";")) !== FALSE) {
+            if (count($line) > 1 && $line[1] === $email) {
+                fclose($file);
+                return [
+                    'name' => $line[0],
+                    'email' => $line[1],
+                    'password' => $line[2]
+                ];
+            }
+        }
+        fclose($file);
+    }
+
+    return null;
+}
+
+function registerNewUser($filename, $name, $email, $password)
+{
+    // Fetch all users
+    $users = fetchUsers($filename);
+
+    // Check if email already exists
+    if (isset($users[$email])) {
+        return 'Email already registered';
+    }
+
+    // Encrypt the password before storing it
+    $hashed_password = password_hash($password, PASSWORD_DEFAULT);
+
+    // Prepare the user data as a CSV string
+    $user_record = implode(";", [$name, $email, $hashed_password]) . PHP_EOL;
+
+    // Append the user record to the file
+    file_put_contents($filename, $user_record, FILE_APPEND);
```

Comment:
po uspesne registraci neni zaznam uzivatele pridan https://eso.vse.cz/~nikd01/cv04/admin/users.php

Reply:
now it should be okay - it wasn't the code related issue, i just didn't set the writing rights for the "database.db" file on the server